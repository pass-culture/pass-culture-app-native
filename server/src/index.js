"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.server = void 0;
const firebase_functions_1 = require("firebase-functions");
const app_1 = require("./app");
const PORT = Number(process.env.PORT) || 8080;
exports.server = app_1.app.listen(PORT, () => {
    /* istanbul ignore next */
    // eslint-disable-next-line no-console
    process.env.NODE_ENV !== 'test' && firebase_functions_1.logger.info(`[${process.env.ENV}] Listening on port ${PORT}`);
});
