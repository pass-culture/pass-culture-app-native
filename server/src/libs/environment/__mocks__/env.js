"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.env = void 0;
exports.env = {
    ENV: 'test',
    APP_PUBLIC_URL: 'http://localhost:8080',
    APP_PROXY_URL: 'https://app-proxy.testing.passculture.team',
    API_BASE_URL: 'https://api.testing.passculture.team',
    API_BASE_PATH_NATIVE_V1: 'native/v1',
    DEEPLINK_PROTOCOL: 'passculture://',
    PROXY_CACHE_CONTROL: 'public,max-age=3600',
};
