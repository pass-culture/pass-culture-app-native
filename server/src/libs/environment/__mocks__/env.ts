import { Environment } from '../types'

export const env: Environment = {
  ENV: 'test',
  APP_PUBLIC_URL: 'http://localhost:8080',
  APP_PROXY_URL: 'https://app-proxy.testing.passculture.team',
  API_BASE_URL: 'https://api.testing.passculture.team',
  API_BASE_PATH_NATIVE_V1: 'native/v1',
  DEEPLINK_PROTOCOL: 'passculture://',
  PROXY_CACHE_CONTROL: 'public,max-age=3600',
}
