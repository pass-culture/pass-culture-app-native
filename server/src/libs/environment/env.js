"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.env = void 0;
const path_1 = require("path");
const dotenv_1 = require("dotenv");
(0, dotenv_1.config)({
    path: (0, path_1.resolve)(__dirname, '../../..', `.env.${process.env.ENV}`),
});
exports.env = process.env;
