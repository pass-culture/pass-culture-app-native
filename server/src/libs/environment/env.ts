import { existsSync } from 'fs'
import { resolve, join } from 'path'

import { config as dotEnvConfig } from 'dotenv'

import { parseBooleanVariables } from './parseBooleanVariables'

const pathDevSource = join(__dirname, '../../..', `.env.${process.env.ENV}`)
const pathBuildSource = join(__dirname, '../../../..', '.env')
const path = existsSync(pathDevSource) ? resolve(pathDevSource) : resolve(pathBuildSource)
const __DEV__ = pathDevSource === path

if (!existsSync(path)) {
  throw new Error(`DotEnv configuration not found for environment ${process.env.ENV} (${path})`)
}

dotEnvConfig({ path })

export const env = parseBooleanVariables({ ...process.env, __DEV__ })
