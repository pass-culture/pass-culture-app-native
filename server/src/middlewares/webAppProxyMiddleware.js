"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.webAppProxyMiddleware = exports.metasResponseInterceptor = void 0;
const firebase_functions_1 = require("firebase-functions");
const http_proxy_middleware_1 = require("http-proxy-middleware");
const env_1 = require("../libs/environment/env");
const types_1 = require("../services/entities/types");
const metas_1 = require("../utils/metas");
const { APP_PROXY_URL } = env_1.env;
const { href } = new URL(APP_PROXY_URL);
const ENTITY_PATH_REGEXP = new RegExp(`/(${Object.keys(types_1.ENTITY_MAP).join('|')})/(\\d+)`);
const options = {
    target: href,
    changeOrigin: true,
    ws: true,
    selfHandleResponse: true,
    onProxyRes: (0, http_proxy_middleware_1.responseInterceptor)(metasResponseInterceptor),
};
function metasResponseInterceptor(responseBuffer, proxyRes, req) {
    return __awaiter(this, void 0, void 0, function* () {
        if (proxyRes.headers['content-type'] !== 'text/html') {
            return responseBuffer;
        }
        const html = responseBuffer.toString('utf8');
        let match;
        /* istanbul ignore next */
        // error with istanbul thinking there is a else path
        if (req.url) {
            match = ENTITY_PATH_REGEXP.exec(req.url);
        }
        const [endpoint, entityKey, id] = match || [];
        if (!id) {
            return html;
        }
        try {
            return (0, metas_1.replaceHtmlMetas)(html, endpoint, entityKey, Number(id));
        }
        catch (error) {
            // FIXME: when replaceHtmlMetas can really throw error, restore coverage for following lines and add a throw error unit test
            /* istanbul ignore next */
            // eslint-disable-next-line no-console
            firebase_functions_1.logger.info(`Replacing HTML metas failed: ${error.message}`);
            /* istanbul ignore next */
            return html;
        }
    });
}
exports.metasResponseInterceptor = metasResponseInterceptor;
exports.webAppProxyMiddleware = (0, http_proxy_middleware_1.createProxyMiddleware)(options);
