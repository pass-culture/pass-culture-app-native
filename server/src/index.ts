// for some reason, this is scanned by root project and cause a linting error
// eslint-disable-next-line import/no-unresolved

import { app } from './app'
import { env } from './libs/environment/env'
import { logger } from './utils/logging'

const PORT = Number(process.env.PORT) || 8080

export const server = app.listen(PORT, () => {
  // eslint-disable-next-line no-console
  /* istanbul ignore next */
  process.env.NODE_ENV !== 'test' && logger.info(`[${env.ENV}] Listening on port ${PORT}`)
})
