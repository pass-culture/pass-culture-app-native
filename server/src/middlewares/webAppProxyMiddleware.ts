import { IncomingMessage, ServerResponse } from 'http'

import { createProxyMiddleware, responseInterceptor } from 'http-proxy-middleware'

import { env } from '../libs/environment/env'
import { ENTITY_MAP, EntityKeys } from '../services/entities/types'
import { logger } from '../utils/logging'
import { replaceHtmlMetas } from '../utils/metas'

const { APP_BUCKET_URL } = env

const { href } = new URL(APP_BUCKET_URL)

const ENTITY_PATH_REGEXP = new RegExp(`/(${Object.keys(ENTITY_MAP).join('|')})/(\\d+)`)

const options = {
  target: href,
  changeOrigin: true,
  ws: true,
  selfHandleResponse: true,
  onProxyRes: responseInterceptor(metasResponseInterceptor),
}

const addCanonicalLinkToHTML = (html: string, url: URL): string => {
  const isThematicHome = url.pathname === '/accueil-thematique'
  const homeIdParams = url.searchParams.get('homeId')
  const additionalParams = isThematicHome && homeIdParams ? `?homeId=${homeIdParams}` : ''
  return html.replace(
    '<head>',
    `<head><link rel="canonical" href="${env.APP_PUBLIC_URL}${url.pathname}${additionalParams}" />`
  )
}

const addNoIndexToHTML = (html: string): string =>
  html.replace('<head>', `<head><meta name="robots" content="noindex" />`)

export async function metasResponseInterceptor(
  responseBuffer: Buffer,
  proxyRes: IncomingMessage,
  req: IncomingMessage,
  res: ServerResponse
) {
  if (proxyRes.headers['content-type'] !== 'text/html') {
    return responseBuffer
  }

  // GCP return 404 when returning the index.html if not the base path, this fix it.
  res.statusCode = 200

  let html = responseBuffer.toString('utf8')

  let match

  /* istanbul ignore next */
  // error with istanbul thinking there is a else path
  if (req.url) {
    match = ENTITY_PATH_REGEXP.exec(req.url)

    const url = new URL(req.url.startsWith('/') ? env.APP_PUBLIC_URL + req.url : req.url)
    html = addCanonicalLinkToHTML(html, url)

    const isSearchPageWithParams =
      url.pathname === '/recherche' && [...url.searchParams.keys()].length
    if (isSearchPageWithParams) {
      html = addNoIndexToHTML(html)
    }
  }

  const [endpoint, entityKey, id] = match || []

  if (!id) {
    return html
  }

  try {
    return await replaceHtmlMetas(html, endpoint, entityKey as EntityKeys, Number(id))
  } catch (error) {
    // FIXME(kopax): when replaceHtmlMetas can really throw error, restore coverage for following lines and add a throw error unit test
    /* istanbul ignore next */
    // eslint-disable-next-line no-console
    logger.info(`Replacing HTML metas failed: ${(error as Error).message}`)
    /* istanbul ignore next */
    return html
  }
}

export const webAppProxyMiddleware = createProxyMiddleware(options)
