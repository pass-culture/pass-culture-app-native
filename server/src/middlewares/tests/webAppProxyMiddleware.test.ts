import { IncomingMessage, ServerResponse } from 'http'

import {
  OFFER_RESPONSE_SNAPSHOT,
  TEST_HTML,
  VENUE_WITH_BANNER_RESPONSE_SNAPSHOT,
  VENUE_WITHOUT_BANNER_RESPONSE_SNAPSHOT,
} from '../../../tests/constants'
import { env } from '../../libs/environment/env'
import { metasResponseInterceptor } from '../webAppProxyMiddleware'
import { server } from '../../../tests/server'

const responseBuffer = Buffer.from(TEST_HTML)
const req = {} as IncomingMessage
const res = {} as ServerResponse

const htmlResponseFor = async (url: string): Promise<string | Buffer> => {
  const proxyRes = {
    headers: {
      'content-type': 'text/html',
    },
  } as IncomingMessage

  return await metasResponseInterceptor(
    responseBuffer,
    proxyRes,
    {
      ...req,
      url,
    } as IncomingMessage,
    res
  )
}

const hasCanonicalWith = (href: string): RegExp => {
  const pattern = `<head>.*?<link rel="canonical" href="${href}" />.*?</head>`
  return new RegExp(pattern, 's')
}

describe('metasResponseInterceptor', () => {
  describe('with mock backend', () => {
    beforeAll(() => {
      server.listen()
    })

    afterAll(() => {
      server.resetHandlers()
      server.close()
    })

    it('should return unmodified response buffer when content-type is NOT text/html', async () => {
      const imagePngResponseBuffer = Buffer.from('I am an image/png')
      const notTextHtmlProxyRes = {
        headers: {
          'content-type': 'image/png',
        },
      } as IncomingMessage

      const unmodifiedResponseBuffer = await metasResponseInterceptor(
        imagePngResponseBuffer,
        notTextHtmlProxyRes,
        req,
        res
      )

      expect(unmodifiedResponseBuffer).toEqual(imagePngResponseBuffer)
    })

    it.each([
      ['offer', 'offre', `${OFFER_RESPONSE_SNAPSHOT.id}`],
      ['venue', 'lieu', `${VENUE_WITH_BANNER_RESPONSE_SNAPSHOT.id}`],
      ['venue', 'lieu', `${VENUE_WITHOUT_BANNER_RESPONSE_SNAPSHOT.id}`],
    ])(
      `should edit html when req.url on %s use valid id: /%s/%s`,
      async (_entity: string, endpoint: string, id: string) => {
        const url = `${env.APP_PUBLIC_URL}/${endpoint}${id ? `/${id}` : ''}`
        const finalResponseBuffer = await htmlResponseFor(url)

        expect(finalResponseBuffer).not.toEqual(responseBuffer.toString('utf8'))
      }
    )

    describe('should add canonical link tag', () => {
      it.each(['/connexion', '/creation-compte'])('when url is %s', async (path) => {
        const url = `${env.APP_PUBLIC_URL}${path}`

        const response = await htmlResponseFor(url)

        expect(response).toMatch(hasCanonicalWith(url))
      })

      it.each(['/connexion', '/creation-compte'])(
        'without query params when url is %s with query params',
        async (path) => {
          const canonical = `${env.APP_PUBLIC_URL}${path}`

          const response = await htmlResponseFor(`${canonical}?truc=toto`)

          expect(response).toMatch(hasCanonicalWith(canonical))
        }
      )

      it.each([
        `/offre/${OFFER_RESPONSE_SNAPSHOT.id}`,
        `/lieu/${VENUE_WITH_BANNER_RESPONSE_SNAPSHOT.id}`,
        `/lieu/${VENUE_WITHOUT_BANNER_RESPONSE_SNAPSHOT.id}`,
      ])('with id when url is %s', async (path) => {
        const url = `${env.APP_PUBLIC_URL}${path}`

        const response = await htmlResponseFor(url)

        expect(response).toMatch(hasCanonicalWith(url))
      })

      it("when using `yarn dev` we don't have the host", async () => {
        const response = await htmlResponseFor('/')

        expect(response).toMatch(hasCanonicalWith(`${env.APP_PUBLIC_URL}/`))
      })
    })
  })

  describe('with real backend', () => {
    // we use the staging backend to ensure that the offer/venue data is not changing at every sandbox reset
    const originalApiBaseUrl = env.API_BASE_URL
    beforeAll(() => {
      env.API_BASE_URL = 'https://backend.staging.passculture.team'
    })
    afterAll(() => {
      env.API_BASE_URL = originalApiBaseUrl
    })

    const responseBuffer = Buffer.from(TEST_HTML)
    const proxyRes = {
      headers: {
        'content-type': 'text/html',
      },
    } as IncomingMessage
    const req = {} as IncomingMessage
    const res = {} as ServerResponse

    it('should request the real testing backend and get the offer data', async () => {
      const url = `${env.APP_PUBLIC_URL}/offre/1`
      const html = await metasResponseInterceptor(
        responseBuffer,
        proxyRes,
        {
          ...req,
          url,
        } as IncomingMessage,
        res
      )

      const htmlWithoutOfferId = html.toString()

      expect(htmlWithoutOfferId).toMatchSnapshot()
    })

    it('should request the real testing backend and get the venue data', async () => {
      const url = `${env.APP_PUBLIC_URL}/lieu/9` // 9 is the id of the first permanent venue in the staging database
      const finalResponseBuffer = await metasResponseInterceptor(
        responseBuffer,
        proxyRes,
        {
          ...req,
          url,
        } as IncomingMessage,
        res
      )

      expect(finalResponseBuffer).toMatchSnapshot()
    })
  })
})
