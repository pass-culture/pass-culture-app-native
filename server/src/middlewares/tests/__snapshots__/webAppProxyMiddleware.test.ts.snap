// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metasResponseInterceptor with real backend should request the real testing backend and get the offer data 1`] = `
"<!doctype html>
<html lang=\\"fr\\">
  <head><link rel=\\"canonical\\" href=\\"http://localhost:8080/offre/1\\" />
    <meta charset=\\"utf-8\\">
    <title>pass Culture</title>
    <meta name=\\"title\\" content=\\"pass Culture | Atelier danses urbaines\\" />
    <meta name=\\"description\\" content=\\"Les danseurs de tout style et de tout niveau, mais aussi tous les curieux, sont invités à participer à cet atelier conçu comme un moment de partage et animé par un danseur professionnel et un dj.\\" />
    <meta name=\\"author\\" content=\\"pass Culture\\"/>
    <meta property=\\"og:type\\" content=\\"website\\"/>
    <meta property=\\"og:url\\" content=\\"http://localhost:8080/offre/1\\" />
    <meta property=\\"og:title\\" content=\\"pass Culture | Atelier danses urbaines\\" />
    <meta property=\\"og:description\\" content=\\"Les danseurs de tout style et de tout niveau, mais aussi tous les curieux, sont invités à participer à cet atelier conçu comme un moment de partage et animé par un danseur professionnel et un dj.\\" />
    <meta property=\\"og:image\\" content=\\"https://storage.googleapis.com/passculture-metier-ehp-staging-assets-fine-grained/thumbs/mediations/HE\\" />
    <meta property=\\"og:image:alt\\" content=\\"Atelier danses urbaines\\" />
    <meta property=\\"og:locale\\" content=\\"fr_FR\\"/>
    <meta property=\\"og:site_name\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:card\\" content=\\"summary\\" />
    <meta name=\\"twitter:url\\" content=\\"http://localhost:8080/offre/1\\" />
    <meta name=\\"twitter:title\\" content=\\"pass Culture | Atelier danses urbaines\\" />
    <meta name=\\"twitter:description\\" content=\\"Les danseurs de tout style et de tout niveau, mais aussi tous les curieux, sont invités à participer à cet atelier conçu comme un moment de partage et animé par un danseur professionnel et un dj.\\" />
    <meta name=\\"twitter:image\\" content=\\"https://storage.googleapis.com/passculture-metier-ehp-staging-assets-fine-grained/thumbs/mediations/HE\\" />
    <meta name=\\"twitter:image:alt\\" content=\\"Atelier danses urbaines\\" />
    <meta name=\\"twitter:site\\" content=\\"@pass_Culture\\"/>
    <meta name=\\"twitter:app:country\\" content=\\"FR\\"/>
    <meta name=\\"twitter:app:name:iphone\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:iphone\\" content=\\"1557887412\\"/>
    <meta name=\\"twitter:app:url:iphone\\" content=\\"passculture://http://localhost:8080/offre/1\\" />
    <meta name=\\"twitter:app:name:ipad\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:ipad\\" content=\\"1557887412\\"/>
    <meta name=\\"twitter:app:url:ipad\\" content=\\"passculture://http://localhost:8080/offre/1\\" />
    <meta name=\\"twitter:app:name:googleplay\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:googleplay\\" content=\\"app.passculture.testing\\"/>
    <meta name=\\"twitter:app:url:googleplay\\" content=\\"passculture://http://localhost:8080/offre/1\\" />
    <meta property=\\"al:ios:app_name\\" content=\\"pass Culture\\"/>
    <meta property=\\"al:ios:app_store_id\\" content=\\"1557887412\\"/>
    <meta property=\\"al:ios:url\\" content=\\"passculture://https://app.testing.passculture.team\\"/>
    <meta property=\\"al:android:url\\" content=\\"passculture://https://app.testing.passculture.team\\">
    <meta property=\\"al:android:app_name\\" content=\\"pass Culture\\"/>
    <meta property=\\"al:android:package\\" content=\\"app.passculture.testing\\"/>
    <script type=\\"application/ld+json\\">{\\"@context\\":\\"https://schema.org\\",\\"@type\\":\\"Event\\",\\"name\\":\\"Atelier danses urbaines\\",\\"description\\":\\"Les danseurs de tout style et de tout niveau, mais aussi tous les curieux, sont invités à participer à cet atelier conçu comme un moment de partage et animé par un danseur professionnel et un dj.\\",\\"image\\":\\"https://storage.googleapis.com/passculture-metier-ehp-staging-assets-fine-grained/thumbs/mediations/HE\\",\\"offers\\":{\\"@type\\":\\"AggregateOffer\\",\\"priceCurrency\\":\\"EUR\\",\\"lowPrice\\":\\"0.00\\"},\\"location\\":{\\"@type\\":\\"Place\\",\\"name\\":\\"Le Centquatre-Paris\\",\\"address\\":{\\"@type\\":\\"PostalAddress\\",\\"streetAddress\\":\\"5 rue Curial\\",\\"postalCode\\":\\"75019\\",\\"addressLocality\\":\\"Paris\\"},\\"geo\\":{\\"@type\\":\\"GeoCoordinates\\",\\"latitude\\":\\"48.89005\\",\\"longitude\\":\\"2.37068\\"}},\\"startDate\\":\\"2018-05-19T14:00\\"}</script>
  </head>
  <body>
    <p>Hello world</p>
  </body>
</html>"
`;

exports[`metasResponseInterceptor with real backend should request the real testing backend and get the venue data 1`] = `
"<!doctype html>
<html lang=\\"fr\\">
  <head><link rel=\\"canonical\\" href=\\"http://localhost:8080/lieu/9\\" />
    <meta charset=\\"utf-8\\">
    <title>pass Culture</title>
    <meta name=\\"title\\" content=\\"pass Culture | Relais culturel La Nef\\" />
    <meta name=\\"description\\"
          content=\\"Dispositif porté par le ministère de la Culture, a pour but de faciliter l’accès des jeunes de 18 ans à la culture en leur offrant un crédit de 300€ à dépenser sur l’application pass Culture.\\"/>
    <meta name=\\"author\\" content=\\"pass Culture\\"/>
    <meta property=\\"og:type\\" content=\\"website\\"/>
    <meta property=\\"og:url\\" content=\\"http://localhost:8080/lieu/9\\" />
    <meta property=\\"og:title\\" content=\\"pass Culture | Relais culturel La Nef\\" />
    <meta property=\\"og:description\\"
          content=\\"Dispositif porté par le ministère de la Culture, a pour but de faciliter l’accès des jeunes de 18 ans à la culture en leur offrant un crédit de 300€ à dépenser sur l’application pass Culture.\\"/>
    <meta property=\\"og:image\\" content=\\"https://storage.googleapis.com/passculture-metier-ehp-staging-assets-fine-grained/assets/venue_default_images/mark-williams-9bNmhMKQM1Q-unsplash_1.png\\" />
    <meta property=\\"og:image:alt\\" content=\\"pass Culture\\"/>
    <meta property=\\"og:locale\\" content=\\"fr_FR\\"/>
    <meta property=\\"og:site_name\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:card\\" content=\\"summary_large_image\\" />
    <meta name=\\"twitter:url\\" content=\\"http://localhost:8080/lieu/9\\" />
    <meta name=\\"twitter:title\\" content=\\"pass Culture | Relais culturel La Nef\\" />
    <meta name=\\"twitter:description\\"
          content=\\"Dispositif porté par le ministère de la Culture, a pour but de faciliter l’accès des jeunes de 18 ans à la culture en leur offrant un crédit de 300€ à dépenser sur l’application pass Culture.\\"/>
    <meta name=\\"twitter:image\\" content=\\"https://storage.googleapis.com/passculture-metier-ehp-staging-assets-fine-grained/assets/venue_default_images/mark-williams-9bNmhMKQM1Q-unsplash_1.png\\" />
    <meta name=\\"twitter:image:alt\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:site\\" content=\\"@pass_Culture\\"/>
    <meta name=\\"twitter:app:country\\" content=\\"FR\\"/>
    <meta name=\\"twitter:app:name:iphone\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:iphone\\" content=\\"1557887412\\"/>
    <meta name=\\"twitter:app:url:iphone\\" content=\\"passculture://http://localhost:8080/lieu/9\\" />
    <meta name=\\"twitter:app:name:ipad\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:ipad\\" content=\\"1557887412\\"/>
    <meta name=\\"twitter:app:url:ipad\\" content=\\"passculture://http://localhost:8080/lieu/9\\" />
    <meta name=\\"twitter:app:name:googleplay\\" content=\\"pass Culture\\"/>
    <meta name=\\"twitter:app:id:googleplay\\" content=\\"app.passculture.testing\\"/>
    <meta name=\\"twitter:app:url:googleplay\\" content=\\"passculture://http://localhost:8080/lieu/9\\" />
    <meta property=\\"al:ios:app_name\\" content=\\"pass Culture\\"/>
    <meta property=\\"al:ios:app_store_id\\" content=\\"1557887412\\"/>
    <meta property=\\"al:ios:url\\" content=\\"passculture://https://app.testing.passculture.team\\"/>
    <meta property=\\"al:android:url\\" content=\\"passculture://https://app.testing.passculture.team\\">
    <meta property=\\"al:android:app_name\\" content=\\"pass Culture\\"/>
    <meta property=\\"al:android:package\\" content=\\"app.passculture.testing\\"/>
  </head>
  <body>
    <p>Hello world</p>
  </body>
</html>"
`;
